src/toil/job.py

```diff
@@ -1480,14 +1499,14 @@ def run(self, fileStore):
         self._rvs = {}  # Set this to avoid the return values being updated after the
         #run method has completed!
         #Now flag that the service is running jobs can connect to it
-        assert self.startFileStoreID != None
-        assert fileStore.jobStore.fileExists(self.startFileStoreID)
-        fileStore.jobStore.deleteFile(self.startFileStoreID)
-        assert not fileStore.jobStore.fileExists(self.startFileStoreID)
+        assert self.startJobStoreID != None
+        if fileStore.jobStore.fileExists(self.startJobStoreID):
+            fileStore.jobStore.deleteFile(self.startJobStoreID)
+        assert not fileStore.jobStore.fileExists(self.startJobStoreID)
         #Now block until we are told to stop, which is indicated by the removal
         #of a file
-        assert self.stopFileStoreID != None
-        while fileStore.jobStore.fileExists(self.stopFileStoreID):
+        assert self.terminateJobStoreID != None
+        while fileStore.jobStore.fileExists(self.terminateJobStoreID):
             time.sleep(1) #Avoid excessive polling
```

Comment:
This should use exponential back-off, see #557. We will hit the request rate limit if we are not careful!!!